### PR TITLE
[FIX] account_edi_*: Fix EDI addresses

### PR DESCRIPTION
**Pull request by aboo-odoo**
* = facturx, ubl, ubl_cii, ubl_bis3

Problem
---------
Currently, the EDI use the commercial partner to craft the XML document. However, this causes issues when users add, for example, an invoice address to a partner. Indeed, the address of the main partner will be used and not the invoice address. This caused issue; see the relevant OPW-3624205.

Objective
---------
Make sure that the correct address is used when generating the XML.

Solution
---------
Make sure the commercial partner // partner value is used at the correct spot.
- `partner` should be used for addresses
- `commercial_partner` for everything else.

OPW-3624205
task-3636315

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/account_edi_ubl_cii/models/account_edi_xml_ubl_20.py [hunk 1]
```diff
@@ -79,9 +79,9 @@ def _get_partner_party_vals(self, partner, role):
         return {
             'partner': partner,
             'party_identification_vals': self._get_partner_party_identification_vals_list(partner),
-            'party_name_vals': [{'name': partner.name}],
+            'party_name_vals': [{'name': partner.display_name}],
             'postal_address_vals': self._get_partner_address_vals(partner),
-            'party_tax_scheme_vals': self._get_partner_party_tax_scheme_vals_list(partner, role),
+            'party_tax_scheme_vals': self._get_partner_party_tax_scheme_vals_list(partner.commercial_partner_id, role),
             'party_legal_entity_vals': self._get_partner_party_legal_entity_vals_list(partner),
```
Comment (JulienVR) [inline addons/account_edi_ubl_cii/models/account_edi_xml_ubl_20.py:85]: I think we should use the commercial_partner_id there as well (and then we can remove the `commercial_partner = partner.commercial_partner_id` in `_get_partner_party_legal_entity_vals_list`)

File: addons/account_edi_ubl_cii/models/account_edi_xml_ubl_bis3.py [hunk 2]
```diff
@@ -111,48 +111,50 @@ def _get_partner_party_vals(self, partner, role):
...
 
     def _get_partner_party_identification_vals_list(self, partner):
         # EXTENDS account.edi.xml.ubl_21
         vals = super()._get_partner_party_identification_vals_list(partner)
 
-        if partner.country_code == 'NL' and 'l10n_nl_oin' in partner._fields:
-            endpoint = partner.l10n_nl_oin or partner.l10n_nl_kvk
+        commercial_partner = partner.commercial_partner_id
+        if commercial_partner.country_code == 'NL' and 'l10n_nl_oin' in commercial_partner._fields:
+            endpoint = commercial_partner.l10n_nl_oin or commercial_partner.l10n_nl_kvk
```
Comment (JulienVR) [inline addons/account_edi_ubl_cii/models/account_edi_xml_ubl_bis3.py:157]: If we're only using the commercial_partner_id in here, isn't easier to just call
`_get_partner_party_identification_vals_list(partner.commercial_partner_id)` ?

**Review by JulienVR — COMMENTED** [review]
Thanks for the PR !

I'm a bit worried that we have to manually take the `commercial_partner_id` in `_get_partner_party_vals` (it's easily forgotten if we extend Bis 3), but I don't really see another way.

Otherwise, it looks good to me.

A detail also: I think the convention of the commit message should be `[FIX] account_edi_*: ...`. Also reference the opw at the end of the commit message :)

File: addons/account_edi_ubl_cii/models/account_edi_xml_ubl_20.py [hunk 3]
```diff
@@ -56,14 +56,12 @@ def _get_partner_party_tax_scheme_vals_list(self, partner, role):
         }]
 
     def _get_partner_party_legal_entity_vals_list(self, partner):
-        commercial_partner = partner.commercial_partner_id
-
         return [{
-            'commercial_partner': commercial_partner,
+            'commercial_partner': partner,
```
Comment (JulienVR) [inline addons/account_edi_ubl_cii/models/account_edi_xml_ubl_20.py:61]: you can remove this ugly space (left by me, oops)

**Review by JulienVR — APPROVED** [review]
LGTM ! Thank you :) 
ping @h4818 / @smetl

File: addons/account_edi_ubl/data/ubl_templates.xml [hunk 4]
```diff
@@ -44,9 +44,9 @@
                         <cbc:Name t-esc="partner.country_id.name"/>
                     </cac:Country>
                 </cac:PostalAddress>
-                <cac:PartyTaxScheme t-if="partner.vat">
-                    <cbc:RegistrationName t-esc="partner.name"/>
-                    <cbc:CompanyID t-esc="partner.vat"/>
+                <cac:PartyTaxScheme t-if="partner.commercial_partner.vat">
```
Comment (h4818) [inline addons/account_edi_ubl/data/ubl_templates.xml:47]: shouldn't it be `partner.commercial_partner_id.vat`

Comment (aboo-odoo) [reply addons/account_edi_ubl/data/ubl_templates.xml:47]: Indeed, mb :+1:

**Comment by h4818** [comment]
Thanks guys! let's go! I haven't tested, but LGTM syntactically
@robodoo r+

**Pull request closed without merging**